compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/ast/scope/Scope.scala

```diff
@@ -0,0 +1,32 @@
+package org.alephium.ralph.lsp.access.compiler.ast.scope
+
+import org.alephium.ralph.{Ast, SourceIndex}
+
+sealed trait Scope {
+  /** The targeted type definition */
+  def typeDef: Ast.Positioned
+
+  /** The scope of the targeted type definition */
+  def scope: SourceIndex
+}
+
+object Scope {
+
+  /** An argument [[Ast.Argument]] and it's scope */
+  case class Argument(typeDef: Ast.Argument,
+                      scope: SourceIndex) extends Scope
+
+  /** A function [[Ast.FuncId]] and it's scope. */
+  case class Function(funcDef: Ast.FuncDef[_],
+                      scope: SourceIndex) extends Scope {
+    /**
+     * When the user clicks on a function call, the targeted AST
+     * will take the user to this [[Ast.FuncId]].
+     *
+     * TODO: Show the entire definition of the function `fn my_func(a: A, b: B)`
+     * to be highlighted, but this information is not available via a single AST type yet.
```

Comment:
insn't [FuncDef ](https://github.com/alephium/dev-alephium/blob/c578ef4fa2db6b34faa9e1df2ddc2d10ae17f2dc/ralph/src/main/scala/org/alephium/ralph/Ast.scala#L680-L690) keeping those informations?

Reply:
This code is removed in one of the newer PRs.

And it isn't. `FuncDef` has a `SourceIndex` that spans the entire function block, including the body.